src/NJsonSchema.CodeGeneration.CSharp/CSharpJsonSerializerGenerator.cs

```diff
@@ -23,47 +23,90 @@ public static string GenerateJsonSerializerParameterCode(CSharpGeneratorSettings
-                           : string.Empty) +
-                       " }" +
-                       (useSettingsTransformationMethod ? ")" : string.Empty);
-            }
-            else
-            {
-                if (hasJsonConverters)
-                {
-                    return ", " + GenerateConverters(jsonConverters);
-                }
-                else
-                {
+                case CSharpJsonLibrary.NewtonsoftJson:
+                    var useSettingsTransformationMethod = !string.IsNullOrEmpty(settings.JsonSerializerSettingsTransformationMethod);
+                    if (settings.HandleReferences || useSettingsTransformationMethod)
+                    {
+                        return ", " +
+                            (useSettingsTransformationMethod ? settings.JsonSerializerSettingsTransformationMethod + "(" : string.Empty) +
+                            "new Newtonsoft.Json.JsonSerializerSettings { " +
+                            (settings.HandleReferences
+                                ? "PreserveReferencesHandling = Newtonsoft.Json.PreserveReferencesHandling.All"
+                                : string.Empty) +
+                            (hasJsonConverters
+                                ? (settings.HandleReferences ? ", " : string.Empty) + "Converters = " + GenerateConverters(jsonConverters, settings.JsonLibrary)
+                                : string.Empty) +
+
+                            // TODO(newtonsoft.json): Add supporting more options for Newtonsoft.Json (hided for System.Text.Json)
+                            // or it can be handled in JsonSerializerSettingsTransformationMethod?
+
+                            " }" +
+                            (useSettingsTransformationMethod ? ")" : string.Empty);
+                    }
+                    else
+                    {
+                        if (hasJsonConverters)
+                        {
+                            return ", " + GenerateConverters(jsonConverters, settings.JsonLibrary);
+                        }
+                        else
+                        {
+                            return string.Empty;
+                        }
+                    }
+                case CSharpJsonLibrary.SystemTextJson:
+                    var useOptionsTransformationMethod = !string.IsNullOrEmpty(settings.JsonSerializerOptionsTransformationMethod);
+                    // TODO: add more conditions?
+                    if (useOptionsTransformationMethod || hasJsonConverters)
+                    {
+                        return ", " +
+                            (useOptionsTransformationMethod ? settings.JsonSerializerOptionsTransformationMethod + "(" : string.Empty) +
+                            "System.Text.Json.JsonSerializerOptions { " +
+
+                            // TODO(system.text.json): Add supporting more options for System.Text.Json (hided for Newtonsoft.Json)
```

Comment:
I think we should only support the existing "high-level" configs which already exists - if you need more you can use JsonSerializerOptionsTransformationMethod

Reply:
Ok, I'll just leave these options.